File: Leetcode_POTD_2025/8_August/3000_Maximum_Area_of_Longest_Diagonal_Rectangle.java
class Solution {
    public int areaOfMaxDiagonal(int[][] dimensions) {
        long longestDiagSq = 0;
        int greatestArea = 0;
        for (int[] entry : dimensions) {
            long currentDiagSq = (long)entry[0] * entry[0] + (long)entry[1] * entry[1];
            int currentArea = entry[0] * entry[1];
            if (currentDiagSq > longestDiagSq) {
                longestDiagSq = currentDiagSq;
                greatestArea = currentArea;
            } else if (currentDiagSq == longestDiagSq && currentArea > greatestArea) {
                greatestArea = currentArea;
            }
        }
        return greatestArea;
    }
}

File: Leetcode_POTD_2025/8_August/3000_Maximum_Area_of_Longest_Diagonal_Rectangle.cpp
class Solution {
public:
    int areaOfMaxDiagonal(vector<vector<int>>& dimensions) {
        long long best_diag_sq = 0;
        int best_region = 0;
        for (auto& dim_pair : dimensions) {
            long long current_diag_sq = static_cast<long long>(dim_pair[0]) * dim_pair[0] + static_cast<long long>(dim_pair[1]) * dim_pair[1];
            int current_region = dim_pair[0] * dim_pair[1];
            if (current_diag_sq > best_diag_sq) {
                best_diag_sq = current_diag_sq;
                best_region = current_region;
            } else if (current_diag_sq == best_diag_sq && current_region > best_region) {
                best_region = current_region;
            }
        }
        return best_region;
    }
};
